### PR TITLE
mock: Make rdkafka_mock.h c++ compatible (#2885)

### PR DESCRIPTION
**Pull request by mpribble**
Allow rdkafka_mock.h to be included from c and c++ programs.

### CODE REVIEW

File: src/rdkafka_mock.h [hunk 1]
```diff
@@ -33,6 +33,9 @@
 #error "rdkafka_mock.h must be included after rdkafka.h"
 #endif
 
+#ifdef __cplusplus
+extern "C" {
+#endif
```
Comment (edenhill) [inline src/rdkafka_mock.h:39]: Can you do the indent trick to keep the IDE from indenting all code to the extern level?
Like this:
https://github.com/edenhill/librdkafka/blob/master/src/rdkafka.h#L53

Comment (mpribble) [reply src/rdkafka_mock.h:42]: Done. I went ahead and squashed the change into the original commit. I **think** that's what your contributor instructions say.

Comment (edenhill) [reply src/rdkafka_mock.h:42]: You're acing this PR thing! 👍

Comment (mpribble) [reply src/rdkafka_mock.h:42]: Thanks the vote of confidence. Luckily I use gitlab on a regular basis, and it's basically a copy/paste of github. Just happy to finally be making my first open source contribution, even a tiny one like this...feels good.

**Comment by edenhill** [comment]
Thank you!

**Pull request merged**